pkg/operators/otel-metrics/otel-metrics.go

```diff
@@ -374,6 +398,39 @@ func (mc *metricsCollector) addValHistFunc(f datasource.FieldAccessor) error {
 	switch f.Type() {
 	default:
 		return fmt.Errorf("unsupported field type for metrics value %q: %s", f.Name(), f.Type())
+	case api.ArrayOf(api.Kind_Uint32):
+		// Calc buckets
+		alen := int(f.Size() / 4)
+
+		bucketVals := make([]int64, alen)
+		bucketValsFloat := make([]float64, alen)
+		for i := range bucketVals {
+			bucketVals[i] = int64(1 << i)
+			bucketValsFloat[i] = float64(bucketVals[i])
+		}
+
+		options = append(options, metric.WithExplicitBucketBoundaries(bucketValsFloat...))
+		hOptions := make([]metric.Int64HistogramOption, len(options))
+		for i, option := range options {
+			hOptions[i] = option
+		}
+
+		hist, err := mc.meter.Int64Histogram(f.Name(), hOptions...)
+		if err != nil {
+			return fmt.Errorf("adding metric histogram for %q: %w", f.Name(), err)
+		}
+		mc.values = append(mc.values, func(ctx context.Context, data datasource.Data, set attribute.Set) {
+			b := f.Get(data)
+			for i := 0; i < alen; i++ {
+				val := binary.LittleEndian.Uint32(b[i*4:])
+				// This looks counterintuitive - and it is. However, there's currently no way to mirror
+				// existing buckets to an otel meter directly.
+				for range val {
+					hist.Record(ctx, bucketVals[i], metric.WithAttributeSet(set))
+				}
```

Comment:
For every entry in the bucket, we have to emit a `hist.Record()` with a value that puts a `+1` into that bucket again. We sadly cannot write directly to the buckets. I plan to look for alternatives to optimize this, but for now this will have to do. I will extend the comment in the code as well.